src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,947 @@
+            return consts.Diagnostic_Check_Failed
+
+        return consts.Diagnostic_Check_Passed
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.MSI_Cert_Check_Fault_Type, summary="Error occurred while trying to perform MSI ceritificate presence check")
+        diagnoser_output.append("An exception has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete
+
+
+def check_cluster_security_policy(corev1_api_instance, helm_client_location, release_namespace):
+
+    global diagnoser_output
+    try:
+        # Intializing the kap_pod_present and cluster_connect_feature variable as False
+        kap_pod_present = False
+        cluster_connect_feature = False
+
+        # CMD command to get helm values in azure arc and converting it to json format
+        command = [helm_client_location, "get", "values", "azure-arc", "--namespace", release_namespace, "-o", "json"]
+
+        # Using Popen to execute the helm get values command and fetching the output
+        response_helm_values_get = Popen(command, stdout=PIPE, stderr=PIPE)
+        output_helm_values_get, error_helm_get_values = response_helm_values_get.communicate()
+        if response_helm_values_get.returncode != 0:
+            if ('forbidden' in error_helm_get_values.decode("ascii") or 'timed out waiting for the condition' in error_helm_get_values.decode("ascii")):
+                telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                        summary='Error while doing helm get values azure-arc')
+
+        # Converting output obtained in json format and fetching the clusterconnect-agent feature
+        helm_values_json = json.loads(output_helm_values_get)
+        cluster_connect_feature = helm_values_json["systemDefaultValues"]["clusterconnect-agent"]["enabled"]
+
+        # To retrieve all of the arc agent pods that are presesnt in the Cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # Traversing thorugh all agents and checking if the Kube aad proxy pod is present or not
+        for each_agent_pod in arc_agents_pod_list.items:
+            if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                kap_pod_present = True
+                break
+
+        # Checking if any pod security policy is set
```

Comment:
change the comment to - checking if there is any chance of pod security policy presence